src/Opdex.Platform.Application/EntryHandlers/Markets/Quotes/CreateSetStandardMarketPermissionsTransactionQuoteCommandHandler.cs

```diff
@@ -26,8 +28,15 @@ public class CreateSetStandardMarketPermissionsTransactionQuoteCommandHandler :
 
         public override async Task<TransactionQuoteDto> Handle(CreateSetStandardMarketPermissionsTransactionQuoteCommand request, CancellationToken cancellationToken)
         {
-            // ensure market exists, if not throw to return 404
-            _ = await _mediator.Send(new RetrieveMarketByAddressQuery(request.Market, findOrThrow: true), cancellationToken);
+            var market = await _mediator.Send(new RetrieveMarketByAddressQuery(request.Market, findOrThrow: true), cancellationToken);
+            if (market.IsStakingMarket) throw new NotFoundException("Market address must represent a standard market.");
```

Comment:
Should this be InvalidDataException instead since the market is found but is the incorrect type? 

Not found suggests the provided market wasn't fouud at all.

Reply:
That's how I originally did it but then I thought 404 makes sense too as the path specifies it is a standard market. Now I'm second guessing that thought.

Reply:
+1 vote for InvalidDataException because the market does exist, just doesn't have that functionality.

Reply:
Updated